developers/release.md

```diff
@@ -0,0 +1,240 @@
+
+```
+git clone https://github.com/Katello/katello.org
+cd katello.org
+git checkout -b KATELLO-X.Y
+git push origin KATELLO-X.Y
+```
+
+Check that the CI job that builds the site (http://ci.theforeman.org/job/deploy_katello_site/) has been kicked off or kick it off manually if needed. Once it passes, visit [http://www.katello.org](http://www.katello.org) and check that X.Y is available from the drop down list. If it did not build correctly, you will need to checkout the `deploy` branch and test it locally.
+
+Congratulations, you have successfully branched all of the repositories and prepared them for release. We recommend double checking each repository ensuring that the branch exists in the upstream repository, that the top of the commit stack in that branch includes the tag you created prior to branching (so that you have an initial tag point that represents the branch point) and that the tag itself exists. The last part of this step is to send to a follow up email letting the community know that everything has been branched.
+
+### Step 5: Configure Koji
+
+Note that to perform this step, Foreman must have generated it's Koji build targets for the targeted Foreman release. This is due to how Katello tags inherit from some Foreman tags. Coordinate with the Foreman release nanny on when they plan to do that step and wait to do this step until then. Koji needs to have configuration added specific to the tags that the release will occur under. The first step is to add new build roots and tags using the following script:
+
+```
+https://github.com/Katello/katello-misc/blob/master/rel-eng/koji_create_buildroots.sh
+```
+
+We recommend running this script in logical chunks instead of attempting to run the entire thing at once.
+
+### Step 6: Setup Mash Scripts
+
+In order to generate the release repositories on Koji, a mash script for the version being released needs to be added. If you do not have direct access to the koji.katello.org box, you can request access or ask someone who does to perform this step for you.
+
+Copy previous mash script to new one
+
+```
+cp /usr/local/bin/katello-mash-split-2.0.py /usr/local/bin/katello-mash-split-2.1.py
+```
+
+Open new mash script and replace the version with the new one
+
+```
+vim /usr/local/bin/katello-mash-split-2.1.py
+:%s/2.0/2.1/g
+```
+
+Now do the same for all of the 2.0 mash files in `/etc/mash`, copying them to 2.1 files and updating the contents from 2.0 to 2.1. Finally, generate the first set of X.Y repositories on koji:
+
+```
+/usr/local/bin/katello-mash-split-2.1.py
+```
+
+The mash can be run via the release_mash_katello Jenkins job. However, the job first needs to be updated for the new version you are building by adding the Foreman and Katello versions to the configuration of the job through the Jenkins Web UI.
+
```

Comment:
I think there's a step missing somewhere in here, as its never explicitly said that one should go into each git repo, tag, and run `tito release koji` with an explanation of what that does.